packages/vm/tests/api/EIPs/eip-2929.spec.ts

```diff
@@ -263,16 +259,34 @@ tape('EIP 2929: gas cost tests', (t) => {
     st.end()
   })
 
-  tape('ensure storage slots are tracked transaction-wide', async (t) => {
+  tape('ensure warm addresses/slots are tracked transaction-wide', async (t) => {
+    // Note: these tests were manually analyzed to check if these are correct.
+    // The gas cost has been taken from these tests.
+
+    // What is covered:
+    // The called address is warm.
+    // (1) Warm storage slots are kept warm during inner calls
+    // (2) If a slot is marked warm, and the inner call reverts, and originally it was cold
+    // then it is still cold.
+    // (1) and (2) are also checked for accounts.
+    // These are manually explicitly checked to ensure the right gas costs are used on
+    // SLOAD or CALL operations.
+
     // load same storage slot twice (also in inner call)
-    //await runCodeTest('60005460003415601357600080808080305AF15B00', 1, t)
+    await runCodeTest('60005460003415601357600080808080305AF15B00', 23369, t)
     // call to contract, load slot 0, revert inner call. load slot 0 in outer call.
-    //await runCodeTest('3415600B57600054600080FD5B600080808080305A60005400', 1, t)
+    await runCodeTest('341515600D57600054600080FD5B600080808080305AF160005400', 25374, t)
 
     // call to address 0xFFFF..FF
-    //const callFF = '6000808080806000195AF1'
-    //await runCodeTest(callFF + '60003415601C57600080808080305AF15B00', 1, t)
-    //await runCodeTest('3415601557' + callFF + '600080FD5B600080808080305A60005400', 1, t)
+    const callFF = '6000808080806000195AF1'
+    // call address 0xFF..FF, now call same contract again, call 0xFF..FF again (it is now warm)
+    await runCodeTest(callFF + '60003415601B57600080808080305AF15B00', 23909, t)
```

Comment:
Out of interest: how did you generate these bytcode sequences? 🤔

Reply:
One of my (I guess) somewhat esoteric hobbies is writing raw EVM assembly. I do this from time to time, and getting more "fluent" in raw EVM 😄  It is very cool to figure out how you can create a minimal implementation which does what it should do. The [minimal proxy contract](https://eips.ethereum.org/EIPS/eip-1167), for instance, is also very interesting. As a side note: I hope that at some point we get a new [gas golf contest](https://g.solidity.cc/). They allow raw assembly too 😄

Reply:
Lol, I absolutely need more hobbies I can brag about

Maybe you can give a course at some PiT 😁  (far from fluent assembly writer)

Reply:
Oh yeah - that sounds fun 😄 , I can definitely prepare something at some PiT 👍